dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/dao/SimpleMongoDBStorageStrategy.java

```diff
@@ -85,14 +89,14 @@ public DBObject getQuery(final RyaStatement stmt) {
         final RyaURI context = stmt.getContext();
         final BasicDBObject query = new BasicDBObject();
         if (subject != null){
-            query.append(SUBJECT, subject.getData());
+            query.append(SUBJECT_HASH, DigestUtils.sha256Hex(subject.getData()));
```

Comment:
I'll do some testing on this, but I'm guessing PRO: (1) smaller index size and (2) smaller messages over the wire.  CON: Need to take care when println'ing the query.